### PR TITLE
Ftp#97 unit tests put

### PR DESCRIPTION
**Pull request by jarfgit**

### CODE REVIEW

File: FTPizza.Tests/FTP.Test.cs [hunk 1]
```diff
@@ -1,11 +1,28 @@
...
+
+        [SetUp]
+        public void Init()
+        {
+            ftpUrl = connectionFile[0].Replace("url=", "");
+            ftpUser = connectionFile[1].Replace("user=", "");
+            ftpPassword = connectionFile[2].Replace("pass=", "");
+            client = new Ftp(ftpUser, ftpPassword, ftpUrl);
+        }
+
```
Comment (kevbui) [inline FTPizza.Tests/FTP.Test.cs]: I really dislike that this unit test has a dependency by interacting with the connectionFile

**Comment by kevbui** [comment]
You should also unit test bad input too
Also all these tests fail

File: FTPizza.Tests/FTP.Test.cs [hunk 2]
```diff
@@ -21,5 +38,30 @@ public void SuccessfulLoginIntoFtpServer()
...
+        public void SuccessfulDeleteDirectoryOnFtpServer()
+        {
+            client.CreateDirectory("UnitTestDirectory");
+
+            string toDelete = "UnitTestDirectory";
+
+            client.DeleteDirectory(toDelete);
+            client.FetchCurrentRemoteDirectoryItems();
+
+            Assert.That(client.currentRemDirFiles.Contains("UnitTestDirectory"), Is.False);
```
Comment (kevbui) [inline FTPizza.Tests/FTP.Test.cs]: It's better to use Assert.That(**_, Has.Member(**_)) instead

File: FTPizza.Tests/FTP.Test.cs (hunk 1 above)
Comment (jarfgit) [inline FTPizza.Tests/FTP.Test.cs]: Matt and I did discuss this yesterday, and agree it isn't the most elegant or correct way to perform a unit test.. However we figured a quick and dirty approach that would make it possible for the rest of the team to write unit tests was preferable given the short amount of time we have left before the assignment is due.

If you have a suggestion to approach this differently, by all means share it. 

We can also put this to a vote...

**Comment by kevbui** [comment]
Well the way I was doing it, was using a anonymous ftp server, and testing it there

**Comment by SoySauceLabs** [comment]
I addressed some of the comments. As for the anonymous server, it deletes files as soon as they're uploaded. So, that server won't help us very much. I could see if our ftp class can be mocked.

**Comment by kevbui** [comment]
Well technically it doesn't matter that the server deletes files after upload, because we still add that file to our currentRemFiles. Where we can check if it exists there.

**Comment by SoySauceLabs** [comment]
Good catch, I'll switch over to the anonymous server then.

File: FTPizza.Tests/FTP.Intergration.Test.cs [hunk 3]
```diff
@@ -28,6 +28,61 @@ public void ShouldDownloadAFIle()
...
+            Assert.That(client, Is.Not.Null);
+        }
+
+        [Test]
+        public void SuccessfulPutFileInToFtpServer()
+        {
+            var upload_client = new Ftp("", "", "speedtest.tele2.net/upload");
+            List<string> uploadList = new List<string>();
+
+            // Make sure file UnitTestFile.txt exists locally in FTPizza\FTPizza.Tests\bin\Debug before executing tests
```
Comment (kevbui) [inline FTPizza.Tests/FTP.Intergration.Test.cs]: Maybe call a streamwriter here to create the file or create one during Setup attribute

**Pull request merged**